src/GraphQL/GraphQLExtensions.cs

```diff
@@ -470,5 +476,40 @@ public static IValue ToAST(this IGraphType type, object value)
 
             throw new ArgumentOutOfRangeException(nameof(type), $"Must provide Input Type, cannot use {type.GetType().Name} '{type}'");
         }
+
+        /// <summary>
+        /// Returns a string representation of the specified node.
+        /// </summary>
+        public static string Print(this ASTNode node)
```

Comment:
Why do we need this?  What is this ASTNode - is it from the parser project?  At first glance, I am not a fan of this method.  The AST printer is now in the Parser project.  Any printing extensions belong there.  Maybe there's a good reason - I need to review further.

Reply:
1. Introspection usage by `__DirectiveArgument` and `__InputValue` to return GraphQL-formatted string representing the default value
2. `SchemaPrinter` that eventually will be completely rewritten
3. `OverlappingFieldsCanBeMerged` uses it to compare arguments (dirty implementation, should be rewritten)
4. `ExecutionHelper` uses it to buil error message: `throw new ArgumentOutOfRangeException(nameof(input), $"Expected object value for '{inputObjectGraphType.Name}', found not an object '{input.Print()}'.");`

Reply:
internal now + rewritten